packages/core/src/common/quick-pick-service.ts

```diff
@@ -101,17 +112,62 @@ export interface QuickPickValue<V> extends QuickPickItem {
 }
 
 export interface QuickInputButton {
-    iconPath?: URI | { light: URI; dark: URI } | { id: string };
+    iconPath?: URI | { light?: URI | Uri; dark: URI | Uri } | { id: string };
```

Comment:
You likely know more than I do about how the QuickPick code should be working. I added this here because there are cases where we declare that we extend Monaco classes, and then we were requiring (Theia) `URI`s, but Monaco complained that that was inconsistent with its typedefs. There may be a more elegant solution, and I'm certainly open to suggestions.

Reply:
I believe that's correct. Since previously, we just wrote whatever we wanted into the declaration file, we may have been fibbing a bit about what types we were actually going to see. Similar issues also came up with `Event` types on our quick-pick delegates. They were trivial, but they indicated that we'd been fudging the types there, too.